### DISCUSSION TITLE
Citations not working for the "h.datasource-azureOpenAI" sample

### DISCUSSION THREAD

**Discussion by JuhoCAO** [OfficeDev/microsoft-365-agents-toolkit#11567] [General]
Hi! I watched this youtube video: https://www.youtube.com/watch?v=VLP32Ci02M0, and tried to recreate it. I got it working, except mine does not show citations. It just shows [doc1] in the end of sentences. Where could the problem be?

![kuva_2024-05-09_132950939](https://github.com/OfficeDev/TeamsFx/assets/144530551/cea3c82b-cc86-4937-b9b5-577d820fab2a)

**Comment by therealjohn** [comment]
@JuhoCAO 

You'll have to use a build of `main` of the `teams-ai` library package - that has the latest changes to format the responses correctly. It will be released soon to NPM. 

OR, you could override the `sayCommand` like:

`app.ai.action<PredictedSayCommand>(...` and basically copy this until it's released: https://github.com/microsoft/teams-ai/blob/main/js/packages/teams-ai/src/actions/SayCommand.ts#L172-L208

Also, you're running the app in the Test Tool, which is not going to show the citations UI like Teams. To see the full experience you will need to run the app in Teams.

**Reply by JuhoCAO** [reply]
Okay, so just integrate that in the app.js? Thank you for answering!
